lib/cfg-block.c

```diff
@@ -121,6 +151,9 @@ cfg_block_generate(CfgBlockGenerator *s, GlobalConfig *cfg, CfgArgs *args, GStri
   GError *error = NULL;
   gchar buf[256];
 
+  if (!_validate_mandatory_options(self->arg_defs, args, reference))
+    return FALSE;
+
   if (!_validate_args(args, self->arg_defs, reference))
     return FALSE;
 
```

Comment:
It's a nitpick but i think mandatory option checking should be part of validate_args. It's getting the same arguments but in a different order.